reconcile/utils/saasherder/saasherder.py

```diff
@@ -405,56 +407,62 @@ def _get_promotion_pubs_and_subs(
+            if not pub_pr:
+                # A subscriber target must have a publisher target
                 self.valid = False
-            else:
-                (pub_saas, pub_rt_name, pub_rt_url) = pub_channel_ref
-
-            for sub_saas, sub_rt_name, sub_rt_url in sub_targets:
-                if not pub_channel_ref:
+            for subs_pr in subs_promotion_refs:
+                if not pub_pr:
                     logging.error(
-                        "Channel is not published by any target\n"
-                        "subscriber_saas: {}\n"
-                        "subscriber_rt: {}\n"
-                        "channel: {}".format(sub_saas, sub_rt_name, sub_channel)
+                        "The subscribed channel is not published by any target\n"
+                        f"subscriber_saas: {subs_pr.saas_file.path}\n"
+                        f"subscriber_rt: {subs_pr.resource_template.name}\n"
+                        f"channel: {subs_channel}"
                     )
                 else:
-                    if sub_rt_url != pub_rt_url:
+                    if subs_pr.resource_template.url != pub_pr.resource_template.url:
                         self.valid = False
                         logging.error(
                             "Subscriber and Publisher targets have different "
                             "source repositories\n"
-                            "publisher_saas: {}\n"
-                            "publisher_rt: {}\n"
-                            "publisher_repo: {}\n"
-                            "subscriber_saas: {}\n"
-                            "subscriber_rt: {}\n"
-                            "subscriber_repo: {}\n".format(
-                                pub_saas,
-                                pub_rt_name,
-                                pub_rt_url,
-                                sub_saas,
-                                sub_rt_name,
-                                sub_rt_url,
-                            )
+                            f"Publisher_saas: {pub_pr.saas_file.path}\n"
+                            f"Publisher_rt: {pub_pr.resource_template.name}\n"
+                            f"Publisher_repo: {pub_pr.resource_template.url}\n"
+                            f"Pubscriber_saas: {subs_pr.saas_file.path}\n"
+                            f"Pubscriber_rt: {subs_pr.resource_template.name}\n"
+                            f"Pubscriber_repo: {subs_pr.resource_template.url}\n"
+                        )
+                    elif (
+                        subs_pr.resource_template_target.namespace.environment
+                        == pub_pr.resource_template_target.namespace.environment
+                        and subs_pr.saas_file.path == pub_pr.saas_file.path
+                    ):
+                        self.valid = False
+                        logging.error(
+                            "Promotions can not happen between targets in the same Saas file "
```

Comment:
Please change the error message slightly. Maybe something like "Promotions can not happen between targets of the same environment in the same Saas file".